front, e2e: add tags for the list of tests

Maymanaf opened the pull request:
I only introduced tags and comments,/some step clean-up in this PR

➡️ It can be discussed that some tests don’t have the `@smoke` tag, unless I missed something on my side.

A new [PR](https://github.com/OpenRailAssociation/osrd/pull/14118) will be created after this one to regroup and rename the test files, following a structure like:
```
tests/
  op/                      
    tabs/
    timetable/
    paced-trains/
    simulation-result/
    round-trips/
    nge/
  stdcm/                   
  utils/                  
  reporter/               
  pages/                 
  assets/                  
```
Within each domain, tests may be incrementally numbered:

```
tests/op/timetable/
  01-train-timetable-filter.spec.ts
  02-train-timetable.spec.ts
  03-train-timetable-multi-selection.spec.ts

```

A new `README.md` file will also be added under /tests to document everything related to the tests. ➡️https://github.com/OpenRailAssociation/osrd/pull/14120

Synar reviewed (commented):
Thanks!

Some general questions:

- the commit seems to do quite a bit more than just add tags. Could you please split the unrelated changes into other commits, or at least explain them in the commit description?

- for many describes, you completely replaced the description of the tests with the tags. Is it intentional?

Maymanaf commented:
> Thanks!
> 
> Some general questions:
> 
> * the commit seems to do quite a bit more than just add tags. Could you please split the unrelated changes into other commits, or at least explain them in the commit description?
> * for many describes, you completely replaced the description of the tests with the tags. Is it intentional?

You’re right, I’ll split the commit to separate the unrelated changes

Regarding the descriptions, yes, that was intentional. The existing test descriptions are already clear enough, and adding just tags in `test.describe` helps indicate the tested theme or feature without needing additional wording

Maymanaf committed:
front: fix descriptions and remove unused comments in e2e tests

Signed-off-by: Maymanaf <med.aymen.naf@gmail.com>

Maymanaf committed:
front: rename scenario and cleanup e2e synchronization test

Signed-off-by: Maymanaf <med.aymen.naf@gmail.com>

Maymanaf committed:
front: split long e2e test and update rolling stock page object

Signed-off-by: Maymanaf <med.aymen.naf@gmail.com>

Maymanaf committed:
front: add explicit tags for test files

Introduce consistent tags across all test files, including the new @smoke tag.
These tags make it easier to run tests by theme and will be important later when
the suite grows, allowing CI to run only the most critical tests efficiently.

Signed-off-by: Maymanaf <med.aymen.naf@gmail.com>

Synar reviewed (approved):
LGTM, thanks!
